src/proc/zero-order.cpp

```diff
@@ -142,15 +144,15 @@ namespace librealsense
         const zero_order_options& options, int zo_point_x, int zo_point_y)
     {
         std::vector<double> rtd(intrinsics.height*intrinsics.width);
-        z2rtd(vertices, rtd.data(), intrinsics, options.baseline);
-        double rtd_zo_value;
+        z2rtd(vertices, rtd.data(), intrinsics, int(options.baseline));
+        double rtd_zo_value; // \\todo Not sure why try_get_zo_rtd_ir_point_values() returns a double and detect_zero_order() uses a float. Which is best representation?
```

Comment:
You can probably change the ```zo_value``` argument (line 118) to double. It should resolve all the warnings about conversions, and make the explicit conversion in line 155 unnecessary.

Reply:
I did this in my forked branch. What's the next step to closing this out?
I'll use the github web page as email replies don't appear to keep the message chain.

Reply:
I checked your changes and everything's fine, just please remove this todo comment and we're good :)

Reply:
I removed todo comment and an unnecessary float cast. Also changed         values.reserve((patch_r + 2ULL) *(patch_r + 2ULL)) so the '+' and '*' don't give precision warning messages.
Final changes are pushed. So if these are accepted this PR and Issue#5733 can be closed.